### PR TITLE
cli: conditionally include yarnPlugin query param in versions:bump upgrade-helper link

### PR DESCRIPTION
**Pull request by mtlewis**
## Hey, I just made a Pull Request!

Depends on https://github.com/backstage/upgrade-helper/pull/16. With this change we automatically link to the right flavor of the diff in upgrade helper based on whether the yarn plugin is installed.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by vinzscam — APPROVED** [review]

**Comment by tylerd-canva** [comment]
Hi there! Just a bystander, but I'm working on https://github.com/backstage/backstage/issues/20658 and it may be a nitpick, but could we consider changing the term `yarnPlugin` to `backstageVersionProtocolSupport` or similar? I've got a branch where I've removed nearly all the explicit references to Yarn and it would be really nice to avoid adding any new ones :)

**Comment by mtlewis** [comment]
Hmm - I've got some reservations about that, to be honest. I think generalizing this parameter bakes in the assumption that we could build a plugin for a different package manager that provides _exactly_ the same functionality. Diffs for the upgrade helper are generated by actually running the yarn plugin so that we can guarantee that the right content is shown for each pair of versions. If there were any differences between the yarn and (say) pnpm implementations, the diff wouldn't reflect them.

With that in mind, if we did have a plugin for pnpm (which by the way would be amazing!), I think I'd probably advocated for us considering something like this:
- Generate pnpm-plugin-based diffs for the upgrade-helper to account for any differences between plugins
- Introduce a new `pnpmPlugin` parameter to the upgrade helper
- Detect use of the pnpm plugin in the versions:bump command [as we do for yarn](https://github.com/backstage/backstage/blob/master/packages/cli/src/commands/versions/bump.ts#L472-L499)
- Factor in the pnpm plugin when deciding whether to use `backstage:^` versions (this also allows us to introduce any necessary variations in output based on which plugin is in use)
- Use the `pnpmPlugin` query param in the versions:bump upgrade helper link when the pnpmPlugin has been detected.

Another option might be to consider something like `?backstageProtocol=true&packageManager=yarn` in the upgrade-helper, but it seems like that would be fine to revisit later if we decide we prefer that to separate `yarnPlugin` & `pnpmPlugin` params.

Despite the wall of text, I definitely don't have a strongly-held viewpoint here - happy to discuss further. @vinzscam / @Rugvip, any thoughts on this?

**Comment by tylerd-canva** [comment]
> Another option might be to consider something like ?backstageProtocol=true&packageManager=yarn in the upgrade-helper, but it seems like that would be fine to revisit later if we decide we prefer that to separate yarnPlugin & pnpmPlugin params.

I actually completely agree with this. It's pretty similar to something I mentioned in my PR:

> * create-app support for choosing your package manager at creation time. but if we have a dynamic template we might also need...
> * backstage-upgrade-helper support. pick your package manager in the UI so that you get an effective diff

Suffice to say, I would not have an objection to leaving this as-is for now.

File: .changeset/swift-laws-applaud.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/cli': patch
+---
+
+Update upgrade-helper link in versions:bump command to include yarnPlugin param when the yarn plugin is installed
```
Comment (Rugvip) [inline .changeset/swift-laws-applaud.md:5]:
```suggestion
Update upgrade-helper link in `versions:bump` command to include `yarnPlugin` param when the yarn plugin is installed
```

**Comment by mtlewis** [comment]
Alright, sounds good @tylerd-canva - appreciate the discussion. The closest thing I've found in pnpm are the [hooks](https://pnpm.io/pnpmfile#hooks) - it looks like you could probably use them for the resolution part, but restoring npm versions during packing might be difficult.

**Review by Rugvip — APPROVED** [review]
Thank you! 👍 

Regarding the yarn plugin discussion, I think the plugin is specific enough for now that we should handle that explicitly as it's being done here

**Pull request merged**